### PR TITLE
Refact gptq to support runing on gaudi

### PR DESCRIPTION
**Pull request by n1ck-guo**
## Type of Change

feature

## Description

Refact gptq to support runing on gaudi

## Expected Behavior & Potential Risk

None

## How has this PR been tested?

local test pass

## Dependency Change?

habana_frameworks

### CODE REVIEW

File: neural_compressor/torch/algorithms/weight_only/gptq.py [hunk 1]
```diff
@@ -930,6 +945,9 @@ def fasterquant(self, W, blocksize=128, percdamp=0.01, groupsize=-1, act_order=F
             Losses[:, i1:i2] = Losses1 / 2
 
             W[:, i2:] -= Err1.matmul(Hinv[i1:i2, i2:])
+            # if using hpu during fasterquant, the need add mark_step here
+            # if "hpu" in self.device:
+            #     htcore.mark_step()
```
Comment (wenhuach21) [inline neural_compressor/torch/algorithms/weight_only/gptq.py:950]: delete useless code

File: neural_compressor/torch/algorithms/weight_only/gptq.py [hunk 2]
```diff
@@ -973,25 +991,27 @@ def free(self):
...
+        # self.maxq = torch.tensor(2**self.bits - 1)
+        self.maxq = 2**self.bits - 1
         self.scheme = "sym" if self.sym else "asym"
         self.double_quant_scheme = "sym" if self.double_quant_sym else "asym"
         self.norm = norm
         self.grid = grid
         self.maxshrink = maxshrink
         if trits:
-            self.maxq = torch.tensor(-1)
+            # self.maxq = torch.tensor(-1)
```
Comment (wenhuach21) [inline neural_compressor/torch/algorithms/weight_only/gptq.py:1009]: delete useless code

**Review by wenhuach21 — APPROVED** [review]

File: neural_compressor/torch/algorithms/weight_only/gptq.py [hunk 3]
```diff
@@ -542,8 +545,12 @@ def forward(layer, *args, **kwargs):
         if self.run_fn:
             if self.run_args:
                 self.run_fn(self.model, *self.run_args)
+                if "hpu" in self.device:
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/gptq.py:548]: How about adding `mark_step` into [Auto_Accelerator](https://github.com/intel/neural-compressor/blob/24419c9044fe227ea806db370c1a30272d026f8a/neural_compressor/torch/utils/auto_accelerator.py#L208)?
We could then directly use `auto_detect_accelerator().mark_step()` without needing to verify the device each time,  also make the code cleaner?

I've applied this approach with `empty_cache()` at HQQ.
https://github.com/intel/neural-compressor/blob/24419c9044fe227ea806db370c1a30272d026f8a/neural_compressor/torch/algorithms/weight_only/hqq/core.py#L156-L157

File: neural_compressor/torch/algorithms/weight_only/gptq.py [hunk 4]
```diff
@@ -30,9 +30,12 @@
 
 from neural_compressor.torch.utils import fetch_module, get_device, logger, set_module
 from neural_compressor.torch.utils.auto_accelerator import auto_detect_accelerator
+from neural_compressor.utils.utility import LazyImport
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/gptq.py:33]: Only `common` and `torch` folders are included in INC3[torch] binary. So `from neural_compressor.utils.xxx import xxx` is not allowed. Let me port the `LazyImport` to the `common/utils`.

Comment (yiliu30) [reply neural_compressor/torch/algorithms/weight_only/gptq.py:33]: Ported `LazyImport` in https://github.com/intel/neural-compressor/pull/1701

File: neural_compressor/torch/algorithms/weight_only/gptq.py (hunk 1 above)
Comment (n1ck-guo) [reply neural_compressor/torch/algorithms/weight_only/gptq.py]: done

File: neural_compressor/torch/algorithms/weight_only/gptq.py (hunk 3 above)
Comment (n1ck-guo) [reply neural_compressor/torch/algorithms/weight_only/gptq.py]: done

File: neural_compressor/torch/algorithms/weight_only/gptq.py (hunk 4 above)
Comment (n1ck-guo) [reply neural_compressor/torch/algorithms/weight_only/gptq.py]: done

**Review by yiliu30 — APPROVED** [review]

**Pull request merged**